presto-kafka/src/main/java/io/prestosql/plugin/kafka/KafkaRecordSet.java

```diff
@@ -269,4 +279,58 @@ public void close()
             kafkaConsumer.close();
         }
     }
+
+    public static FieldValueProvider headerMapValueProvider(MapType varcharMapType, Headers headers)
+    {
+        final Type keyType = varcharMapType.getTypeParameters().get(0);
+        final Type valueArrayType = varcharMapType.getTypeParameters().get(1);
+        final Type valueType = valueArrayType.getTypeParameters().get(0);
+
+        final BlockBuilder mapBlockBuilder = varcharMapType.createBlockBuilder(null, 1);
+        final BlockBuilder builder = mapBlockBuilder.beginBlockEntry();
```

Comment:
Per code style, we do not use `final` for local variables.
For short methods, it doesn't add to readability, and we strive to avoid long methods.

Reply:
Thats fine.